handlers/ocr-handler/server.py

```diff
@@ -0,0 +1,149 @@
+    }
+    
+    # Get request data
+    content = request.get_json()
+    
+    with open('./schemas/request.schema.json') as jsonfile:
+        request_schema = json.load(jsonfile)
+    # Validate incoming request
+    resolver = jsonschema.RefResolver.from_schema(
+        request_schema, store=schema_store)
+    try:
+        validator = jsonschema.Draft7Validator(
+            request_schema,
+            resolver=resolver
+        )
+        validator.validate(content)
+    except jsonschema.exceptions.ValidationError as error:
+        logging.error(error)
+        return jsonify("Invalid Request JSON format"), 400
+    
+    # Check preprocessor data
+    preprocessors = content['preprocessors']
+    
+    # No OCR preprocessor
+    if 'ca.mcgill.a11y.image.preprocessor.ocr' not in preprocessors:
+        logging.error("No OCR preprocessor found")
+        response = {
+            "request_uuid": content["request_uuid"],
+            "timestamp": int(time.time()),
+            "renderings": []
+        }
+        try:
+            validator = jsonschema.Draft7Validator(schema, resolver=resolver)
+            validator.validate(response)
+        except jsonschema.exceptions.ValidationError as error:
+            logging.error(error)
+            return jsonify("Invalid Preprocessor JSON format"), 500
+        logging.debug("Sending response")
+        return response
+    
+    ocr_data = preprocessors['ca.mcgill.a11y.image.preprocessor.ocr']
+    
+    # OCR lines empty
+    if len(ocr_data['lines']) == 0:
+        logging.error("OCR lines empty")
+        response = {
+            "request_uuid": content["request_uuid"],
+            "timestamp": int(time.time()),
+            "renderings": []
+        }
+        try:
+            validator = jsonschema.Draft7Validator(schema, resolver=resolver)
+            validator.validate(response)
+        except jsonschema.exceptions.ValidationError as error:
+            logging.error(error)
+            return jsonify("Invalid Preprocessor JSON format"), 500
+        logging.debug("Sending response")
+        return response
+
+    # Text renderer not supported
+    if 'ca.mcgill.a11y.image.renderer.Text' not in content['renderers']:
+        logging.error("Text renderer not supported")
+        response = {
+            "request_uuid": content["request_uuid"],
+            "timestamp": int(time.time()),
+            "renderings": []
+        }
+        try:
+            validator = jsonschema.Draft7Validator(schema, resolver=resolver)
+            validator.validate(response)
+        except jsonschema.exceptions.ValidationError as error:
+            logging.error(error)
+            return jsonify("Invalid Preprocessor JSON format"), 500
+        logging.debug("Sending response")
+        return response
+    
+    # Get text renderer data
+    text = 'The following ' + str(len(ocr_data['lines'])) + ' lines of text were found in the image: '
```

Comment:
Unsure if we want to adjust "image" to "graphic" or "photo" as we've done elsewhere. I'm ambivalent about that and will just ping @Cybernide in case she has any preference.